pkg/operator/k8sutil/node.go

```diff
@@ -84,8 +88,10 @@ func GetValidNodes(rookStorage rookalpha.StorageScopeSpec, clientset kubernetes.
 			validK8sNodes = append(validK8sNodes, n)
 		}
 	}
+	// Get the corresponding rookv1alpha2.Nodes and update them.
+	rookNodes, validNodes := RookNodesMatchingKubernetesNodes(rookStorage, validK8sNodes)
 
-	return RookNodesMatchingKubernetesNodes(rookStorage, validK8sNodes)
+	return rookNodes, validNodes
```

Comment:
Why are you returning `rookStorage.Nodes`? I don't see it used anywhere where this function is called.

Reply:
Actually, they're both being used later on, and this is the only entry point for matching them with their k8s counterparts. (worth a refactor later, but the library is used in a lot of places). Somewhere along the callchain it's the Location of `rookStorage.Nodes` that's being used for crush.

Reply:
I did try just changing the everything to use ValidStorage, but that seemed to break something and the OSDs deployments wouldn't get created. I think we definitely need a study/refactor of all the assumptions of where the data will be and what triggers what.